Server/ConcordiaCurriculumManager/Security/Requirements/Handlers/GroupMasterHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿using ConcordiaCurriculumManager.Services;
+using Microsoft.AspNetCore.Authorization;
+
+namespace ConcordiaCurriculumManager.Security.Requirements.Handlers;
+
+public class GroupMasterHandler : AuthorizationHandler<GroupMasterOrAdminRequirement>
+{
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly ILogger<GroupMasterHandler> _logger;
+    private readonly IGroupService _groupService;
+
+    public GroupMasterHandler(ILogger<GroupMasterHandler> logger, IGroupService groupService, IHttpContextAccessor contextAccessor)
+    {
+        _logger = logger;
+        _groupService = groupService;
+        _httpContextAccessor = contextAccessor;
+    }
+
+    protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context, GroupMasterOrAdminRequirement requirement)
+    {
+        _logger.LogInformation("Evaluting GroupMasterHanlder");
+
+        if (_httpContextAccessor.HttpContext is null
+            || !_httpContextAccessor.HttpContext.Request.RouteValues.TryGetValue("groupId", out var groupId)
+            || !Guid.TryParse(groupId?.ToString(), out var parsedGroupId))
+        {
+            // This is not an Http Request or there is no group Id. Abstain
+            _logger.LogWarning("GroupMasterHanlder is possibly called on a http endpoint that does not include a group id as a param");
+            return;
+        }
+
+        if (context.User.Identity is null || !context.User.Identity.IsAuthenticated)
```

Comment:
As with the AdminHandler, you can use userAuthenticationService to just extract the ID

Reply:
I mean replacing this `if` statement with `_userAuthenticationService` would just make me use a `try-catch` block since the authorization handlers can be called when the user is not logged in and thus `_httpContextAccessor.HttpContext.User.Claims` would be null.